ptcp performance improvements

janmejay opened the pull request:
- io-work assignment model made #workers independent (earlier it was O(workers)), better work-starvation management
- fully concurrent poller and io-workers (no waiting for io-workers to be done before next poll)
- better message-content handing when reading inbound messages for octate-counted messages
- removed upper-limit on #workers
- enhanced tcpflood to allow producing octate-counted frames, some improvement in error-reporting
- some aesthetic changes
- created a new config-param (processOnPoller, which defaults to "on" in favor of backward comparability) and when turned off, doesn't dispatch any io-work on poller (leaving it completely free to just poll, which makes a lot of difference in maintaining high-throughput with low-latency). Will send a separate PR for documentation-update.

Note: the race-condition fixed in the last patch is not really an old-bug (it couldn't have affected us unless poller was truly running concurrenly to allow a race with io-workers, which was not the case earlier, so it wasn't an old bug really, it was old code which would become buggy if used in a concurrent poller and io-worker setup).

Will add data from benchmarks in comments.

janmejay commented:
Summary

    ** Throughput (messages recvd in 5 seconds, source: impstats):
    *** BEFORE
    **** octate-counted
    ***** mean: 28940.13
    ***** sd: 26085.36
    **** octate-stuffed
    ***** mean: 26702.98
    ***** sd: 23014.67
    
    *** AFTER
    **** octate-counted
    ***** mean: 106286.9
    ***** sd: 10917.69
    **** octate-stuffed
    ***** mean: 94583.81
    ***** sd: 8920.291
    
    ** Latency as mesured by ((10 nodes concurrently running tcpflood -c20 -m100000 -r -d10000 -P129 -D -t 10.34.117.153 -R 1 -S 0):
    *** BEFORE
    **** octate-counted
    sd: 12.25856
    mean: 147.5248
    %iles:
         50%      85%      95%      99%     100%
    149.9530 161.1706 161.9212 161.9402 161.9450
    **** octate-stuffed
    sd: 5.436359
    mean: 138.554
    %iles:
         50%      85%      95%      99%     100%
    138.7230 143.9907 146.8335 147.2331 147.3330
    
    *** AFTER
    **** octate-counted
    sd: 1.551824
    mean: 42.7586
    %iles:
         50%      85%      95%      99%     100%
    42.27700 44.41900 44.94310 45.16702 45.22300
    **** octate-stuffed
    sd: 3.129123
    mean: 46.1467
    %iles:
         50%      85%      95%      99%     100%
    47.21250 47.82500 48.70075 49.17775 49.29700

janmejay commented:
![screenshot-20151027 214438](https://cloud.githubusercontent.com/assets/12777/10785740/070391d2-7d8c-11e5-8ceb-dcf8d658de69.png)
![screenshot-20151027 212831](https://cloud.githubusercontent.com/assets/12777/10785738/07006110-7d8c-11e5-8c90-a8407ebda72c.png)
![screenshot-20151027 212343](https://cloud.githubusercontent.com/assets/12777/10785739/070222c0-7d8c-11e5-88d7-a5a92f692e3b.png)
![screenshot-20151027 151227](https://cloud.githubusercontent.com/assets/12777/10785741/07065188-7d8c-11e5-94f7-2c3e4e6f62c6.png)

janmejay commented:
Github doesn't let me attach text files. I'll try posting inline in next comment details (from which the summary stats are built)

janmejay commented:
* Details

** pre

*** stats with octate counting

Tue Oct 27 14:55:34 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=0 bytes.received=0 bytes.decompressed=0
Tue Oct 27 14:55:39 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=0 bytes.received=0 bytes.decompressed=0
Tue Oct 27 14:55:44 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=21015 bytes.received=106353429 bytes.decompressed=0
Tue Oct 27 14:55:49 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=82361 bytes.received=416715087 bytes.decompressed=0
Tue Oct 27 14:55:54 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=81154 bytes.received=408665508 bytes.decompressed=0
Tue Oct 27 14:55:59 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=79291 bytes.received=402678566 bytes.decompressed=0
Tue Oct 27 14:56:04 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=83191 bytes.received=417452561 bytes.decompressed=0
Tue Oct 27 14:56:09 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=81618 bytes.received=412997058 bytes.decompressed=0
Tue Oct 27 14:56:14 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=68507 bytes.received=347514575 bytes.decompressed=0
Tue Oct 27 14:56:19 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=12571 bytes.received=63688279 bytes.decompressed=0
Tue Oct 27 14:56:24 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=27933 bytes.received=140967938 bytes.decompressed=0
Tue Oct 27 14:56:29 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=35086 bytes.received=177339238 bytes.decompressed=0
Tue Oct 27 14:56:34 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=13997 bytes.received=70852029 bytes.decompressed=0
Tue Oct 27 14:56:39 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=37070 bytes.received=187637505 bytes.decompressed=0
Tue Oct 27 14:56:44 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=33522 bytes.received=168437987 bytes.decompressed=0
Tue Oct 27 14:56:49 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=11477 bytes.received=58152654 bytes.decompressed=0
Tue Oct 27 14:56:54 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=24433 bytes.received=123113794 bytes.decompressed=0
Tue Oct 27 14:56:59 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=21801 bytes.received=109879439 bytes.decompressed=0
Tue Oct 27 14:57:04 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=3532 bytes.received=17456603 bytes.decompressed=0
Tue Oct 27 14:57:09 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=9802 bytes.received=49552913 bytes.decompressed=0
Tue Oct 27 14:57:14 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=6337 bytes.received=31850319 bytes.decompressed=0
Tue Oct 27 14:57:19 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=17482 bytes.received=88166232 bytes.decompressed=0
Tue Oct 27 14:57:24 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=11056 bytes.received=56181650 bytes.decompressed=0
Tue Oct 27 14:57:29 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=0 bytes.received=0 bytes.decompressed=0
Tue Oct 27 14:57:34 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=8873 bytes.received=44848172 bytes.decompressed=0
Tue Oct 27 14:57:39 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=0 bytes.received=0 bytes.decompressed=0
Tue Oct 27 14:57:44 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=15924 bytes.received=79738806 bytes.decompressed=0
Tue Oct 27 14:57:49 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=15198 bytes.received=76434280 bytes.decompressed=0
Tue Oct 27 14:57:54 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=79877 bytes.received=404156201 bytes.decompressed=0
Tue Oct 27 14:57:59 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=0 bytes.received=0 bytes.decompressed=0
Tue Oct 27 14:58:04 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=82203 bytes.received=415545414 bytes.decompressed=0
Tue Oct 27 14:58:09 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=4221 bytes.received=21195224 bytes.decompressed=0
Tue Oct 27 14:58:14 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=70988 bytes.received=358663126 bytes.decompressed=0
Tue Oct 27 14:58:19 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=12865 bytes.received=65049264 bytes.decompressed=0
Tue Oct 27 14:58:24 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=58855 bytes.received=298148123 bytes.decompressed=0
Tue Oct 27 14:58:29 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=8275 bytes.received=42444853 bytes.decompressed=0
Tue Oct 27 14:58:34 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=25788 bytes.received=130509087 bytes.decompressed=0
Tue Oct 27 14:58:39 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=41120 bytes.received=208417014 bytes.decompressed=0
Tue Oct 27 14:58:44 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=5921 bytes.received=29860800 bytes.decompressed=0
Tue Oct 27 14:58:49 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=50222 bytes.received=254659673 bytes.decompressed=0
Tue Oct 27 14:58:54 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=9159 bytes.received=46497685 bytes.decompressed=0
Tue Oct 27 14:58:59 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=28797 bytes.received=146027671 bytes.decompressed=0
Tue Oct 27 14:59:04 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=32000 bytes.received=162530608 bytes.decompressed=0
Tue Oct 27 14:59:09 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=4810 bytes.received=24650551 bytes.decompressed=0
Tue Oct 27 14:59:14 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=31355 bytes.received=159006200 bytes.decompressed=0
Tue Oct 27 14:59:19 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=30618 bytes.received=154843399 bytes.decompressed=0
Tue Oct 27 14:59:25 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=5412 bytes.received=27720939 bytes.decompressed=0
Tue Oct 27 14:59:30 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=12596 bytes.received=63090546 bytes.decompressed=0
Tue Oct 27 14:59:35 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=8969 bytes.received=45390465 bytes.decompressed=0
Tue Oct 27 14:59:40 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=21240 bytes.received=108021715 bytes.decompressed=0
Tue Oct 27 14:59:45 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=25118 bytes.received=127112094 bytes.decompressed=0
Tue Oct 27 14:59:50 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=5110 bytes.received=25945254 bytes.decompressed=0
Tue Oct 27 14:59:55 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=32430 bytes.received=163939178 bytes.decompressed=0
Tue Oct 27 15:00:00 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=30476 bytes.received=154047340 bytes.decompressed=0
Tue Oct 27 15:00:05 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=12171 bytes.received=61550132 bytes.decompressed=0

**** latency for pssh -h /tmp/ips -t 3600 -p 200 -P "time /tmp/tcpflood -c20 -m100000 -r -d10000 -P129 -D -t 10.34.117.153 -R 1 -S 0 -O"
129.485
131.127
138.691
139.594
147.502
152.404
152.777
159.831
161.945
161.892



*** stats with octate stuffing

Tue Oct 27 15:06:05 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=0 bytes.received=0 bytes.decompressed=0
Tue Oct 27 15:06:10 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=0 bytes.received=0 bytes.decompressed=0
Tue Oct 27 15:06:15 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=25482 bytes.received=129425082 bytes.decompressed=0
Tue Oct 27 15:06:20 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=82722 bytes.received=420506036 bytes.decompressed=0
Tue Oct 27 15:06:25 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=68642 bytes.received=351730039 bytes.decompressed=0
Tue Oct 27 15:06:31 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=15436 bytes.received=79116593 bytes.decompressed=0
Tue Oct 27 15:06:36 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=5571 bytes.received=28315987 bytes.decompressed=0
Tue Oct 27 15:06:41 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=5924 bytes.received=29979991 bytes.decompressed=0
Tue Oct 27 15:06:46 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=15260 bytes.received=77532028 bytes.decompressed=0
Tue Oct 27 15:06:51 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=57066 bytes.received=288489344 bytes.decompressed=0
Tue Oct 27 15:06:56 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=72329 bytes.received=366177473 bytes.decompressed=0
Tue Oct 27 15:07:01 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=11296 bytes.received=56609312 bytes.decompressed=0
Tue Oct 27 15:07:06 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=18408 bytes.received=93428886 bytes.decompressed=0
Tue Oct 27 15:07:11 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=7602 bytes.received=37846384 bytes.decompressed=0
Tue Oct 27 15:07:16 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=15927 bytes.received=80340062 bytes.decompressed=0
Tue Oct 27 15:07:21 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=40326 bytes.received=205632435 bytes.decompressed=0
Tue Oct 27 15:07:26 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=10217 bytes.received=52181969 bytes.decompressed=0
Tue Oct 27 15:07:31 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=55366 bytes.received=281340439 bytes.decompressed=0
Tue Oct 27 15:07:36 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=14842 bytes.received=75094189 bytes.decompressed=0
Tue Oct 27 15:07:41 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=23277 bytes.received=118018032 bytes.decompressed=0
Tue Oct 27 15:07:46 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=22615 bytes.received=114607445 bytes.decompressed=0
Tue Oct 27 15:07:51 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=25851 bytes.received=130474855 bytes.decompressed=0
Tue Oct 27 15:07:56 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=14124 bytes.received=71794463 bytes.decompressed=0
Tue Oct 27 15:08:01 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=40089 bytes.received=202406572 bytes.decompressed=0
Tue Oct 27 15:08:06 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=4051 bytes.received=20354201 bytes.decompressed=0
Tue Oct 27 15:08:11 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=12664 bytes.received=63646402 bytes.decompressed=0
Tue Oct 27 15:08:16 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=2867 bytes.received=14924440 bytes.decompressed=0
Tue Oct 27 15:08:21 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=67308 bytes.received=339741419 bytes.decompressed=0
Tue Oct 27 15:08:26 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=50555 bytes.received=255978094 bytes.decompressed=0
Tue Oct 27 15:08:31 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=20959 bytes.received=106236238 bytes.decompressed=0
Tue Oct 27 15:08:36 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=5783 bytes.received=29252392 bytes.decompressed=0
Tue Oct 27 15:08:41 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=9014 bytes.received=45528909 bytes.decompressed=0
Tue Oct 27 15:08:46 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=17074 bytes.received=86883129 bytes.decompressed=0
Tue Oct 27 15:08:51 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=41621 bytes.received=211959462 bytes.decompressed=0
Tue Oct 27 15:08:56 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=71546 bytes.received=362551792 bytes.decompressed=0
Tue Oct 27 15:09:01 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=3246 bytes.received=16594314 bytes.decompressed=0
Tue Oct 27 15:09:06 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=26075 bytes.received=132770220 bytes.decompressed=0
Tue Oct 27 15:09:11 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=7155 bytes.received=36927587 bytes.decompressed=0
Tue Oct 27 15:09:16 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=17790 bytes.received=90399135 bytes.decompressed=0
Tue Oct 27 15:09:21 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=1598 bytes.received=8264195 bytes.decompressed=0
Tue Oct 27 15:09:26 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=67828 bytes.received=345433351 bytes.decompressed=0
Tue Oct 27 15:09:31 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=23186 bytes.received=116697508 bytes.decompressed=0
Tue Oct 27 15:09:37 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=28173 bytes.received=143300345 bytes.decompressed=0
Tue Oct 27 15:09:42 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=19367 bytes.received=98722970 bytes.decompressed=0
Tue Oct 27 15:09:47 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=0 bytes.received=0 bytes.decompressed=0
Tue Oct 27 15:09:52 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=20430 bytes.received=104405214 bytes.decompressed=0
Tue Oct 27 15:09:57 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=57570 bytes.received=291819890 bytes.decompressed=0
Tue Oct 27 15:10:02 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=4105 bytes.received=20726905 bytes.decompressed=0

**** latency for pssh -h /tmp/ips -t 3600 -p 200 -P "time /tmp/tcpflood -c20 -m100000 -r -d10000 -P129 -D -t 10.34.117.153 -R 1 -S 0"
128.426
134.586
136.445
136.236
138.858
139.000
138.588
139.845
146.223
147.333

** post

*** stats with octate stuffing

Tue Oct 27 21:14:18 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=0 bytes.received=0 bytes.decompressed=0
Tue Oct 27 21:14:23 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=0 bytes.received=0 bytes.decompressed=0
Tue Oct 27 21:14:28 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=4225 bytes.received=21200181 bytes.decompressed=0
Tue Oct 27 21:14:33 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=95928 bytes.received=483830079 bytes.decompressed=0
Tue Oct 27 21:14:38 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=102871 bytes.received=520613805 bytes.decompressed=0
Tue Oct 27 21:14:43 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=104687 bytes.received=528207935 bytes.decompressed=0
Tue Oct 27 21:14:48 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=100018 bytes.received=504534662 bytes.decompressed=0
Tue Oct 27 21:14:53 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=98949 bytes.received=502108059 bytes.decompressed=0
Tue Oct 27 21:14:58 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=102889 bytes.received=520650937 bytes.decompressed=0
Tue Oct 27 21:15:03 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=102065 bytes.received=516846429 bytes.decompressed=0
Tue Oct 27 21:15:08 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=100586 bytes.received=509563891 bytes.decompressed=0
Tue Oct 27 21:15:13 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=106295 bytes.received=535449281 bytes.decompressed=0
Tue Oct 27 21:15:18 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=101736 bytes.received=512544890 bytes.decompressed=0
Tue Oct 27 21:15:23 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=101281 bytes.received=510095343 bytes.decompressed=0
Tue Oct 27 21:15:28 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=104254 bytes.received=526140054 bytes.decompressed=0
Tue Oct 27 21:15:33 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=106378 bytes.received=537289259 bytes.decompressed=0
Tue Oct 27 21:15:38 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=105234 bytes.received=530946043 bytes.decompressed=0
Tue Oct 27 21:15:43 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=105788 bytes.received=533615203 bytes.decompressed=0
Tue Oct 27 21:15:48 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=105238 bytes.received=531427325 bytes.decompressed=0
Tue Oct 27 21:15:53 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=107696 bytes.received=543534956 bytes.decompressed=0
Tue Oct 27 21:15:58 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=103477 bytes.received=521936558 bytes.decompressed=0
Tue Oct 27 21:16:03 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=96875 bytes.received=488591757 bytes.decompressed=0
Tue Oct 27 21:16:08 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=80959 bytes.received=408998930 bytes.decompressed=0
Tue Oct 27 21:16:13 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=82979 bytes.received=418383182 bytes.decompressed=0
Tue Oct 27 21:16:18 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=81434 bytes.received=410858110 bytes.decompressed=0
Tue Oct 27 21:16:23 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=81229 bytes.received=408557465 bytes.decompressed=0
Tue Oct 27 21:16:28 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=84868 bytes.received=428078514 bytes.decompressed=0
Tue Oct 27 21:16:33 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=86382 bytes.received=434773625 bytes.decompressed=0
Tue Oct 27 21:16:38 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=84094 bytes.received=424043637 bytes.decompressed=0
Tue Oct 27 21:16:43 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=88818 bytes.received=445732474 bytes.decompressed=0
Tue Oct 27 21:16:48 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=83981 bytes.received=422881374 bytes.decompressed=0
Tue Oct 27 21:16:53 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=89763 bytes.received=451880332 bytes.decompressed=0
Tue Oct 27 21:16:58 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=87623 bytes.received=441480648 bytes.decompressed=0
Tue Oct 27 21:17:03 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=88102 bytes.received=443325583 bytes.decompressed=0
Tue Oct 27 21:17:08 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=101241 bytes.received=511609831 bytes.decompressed=0
Tue Oct 27 21:17:13 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=102821 bytes.received=519040315 bytes.decompressed=0
Tue Oct 27 21:17:18 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=102972 bytes.received=519132719 bytes.decompressed=0
Tue Oct 27 21:17:23 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=100331 bytes.received=504961272 bytes.decompressed=0
Tue Oct 27 21:17:28 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=102850 bytes.received=517630180 bytes.decompressed=0
Tue Oct 27 21:17:33 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=100576 bytes.received=505573258 bytes.decompressed=0
Tue Oct 27 21:17:38 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=104249 bytes.received=526613381 bytes.decompressed=0
Tue Oct 27 21:17:43 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=85904 bytes.received=433875845 bytes.decompressed=0
Tue Oct 27 21:17:48 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=85973 bytes.received=433128422 bytes.decompressed=0
Tue Oct 27 21:17:53 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=84445 bytes.received=427071469 bytes.decompressed=0
Tue Oct 27 21:17:58 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=86822 bytes.received=438524188 bytes.decompressed=0
Tue Oct 27 21:18:03 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=83783 bytes.received=423298169 bytes.decompressed=0
Tue Oct 27 21:18:08 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=83104 bytes.received=419618010 bytes.decompressed=0
Tue Oct 27 21:18:13 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=83747 bytes.received=421863189 bytes.decompressed=0
Tue Oct 27 21:18:18 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=82732 bytes.received=416447253 bytes.decompressed=0
Tue Oct 27 21:18:23 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=83231 bytes.received=418710097 bytes.decompressed=0
Tue Oct 27 21:18:28 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=81023 bytes.received=407926491 bytes.decompressed=0
Tue Oct 27 21:18:33 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=86338 bytes.received=434061339 bytes.decompressed=0
Tue Oct 27 21:18:38 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=79873 bytes.received=402115346 bytes.decompressed=0
Tue Oct 27 21:18:43 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=100481 bytes.received=504834336 bytes.decompressed=0
Tue Oct 27 21:18:48 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=98643 bytes.received=498498977 bytes.decompressed=0
Tue Oct 27 21:18:53 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=101225 bytes.received=508215233 bytes.decompressed=0
Tue Oct 27 21:18:58 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=101095 bytes.received=512548190 bytes.decompressed=0
Tue Oct 27 21:19:03 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=101457 bytes.received=513704546 bytes.decompressed=0
Tue Oct 27 21:19:08 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=104863 bytes.received=530930435 bytes.decompressed=0
Tue Oct 27 21:19:13 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=103790 bytes.received=527733723 bytes.decompressed=0
Tue Oct 27 21:19:18 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=105485 bytes.received=535936191 bytes.decompressed=0
Tue Oct 27 21:19:23 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=103404 bytes.received=519908805 bytes.decompressed=0
Tue Oct 27 21:19:28 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=101161 bytes.received=511649835 bytes.decompressed=0
Tue Oct 27 21:19:33 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=98573 bytes.received=495643687 bytes.decompressed=0
Tue Oct 27 21:19:38 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=86270 bytes.received=433395820 bytes.decompressed=0
Tue Oct 27 21:19:43 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=87608 bytes.received=439840524 bytes.decompressed=0
Tue Oct 27 21:19:48 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=86873 bytes.received=437315540 bytes.decompressed=0
Tue Oct 27 21:19:53 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=88176 bytes.received=445427218 bytes.decompressed=0
Tue Oct 27 21:19:58 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=91727 bytes.received=463851260 bytes.decompressed=0
Tue Oct 27 21:20:03 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=85871 bytes.received=436088295 bytes.decompressed=0
Tue Oct 27 21:20:08 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=88112 bytes.received=446151100 bytes.decompressed=0
Tue Oct 27 21:20:13 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=90166 bytes.received=456321223 bytes.decompressed=0
Tue Oct 27 21:20:18 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=85432 bytes.received=432137740 bytes.decompressed=0
Tue Oct 27 21:20:23 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=86128 bytes.received=433723913 bytes.decompressed=0
Tue Oct 27 21:20:28 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=86496 bytes.received=435546641 bytes.decompressed=0
Tue Oct 27 21:20:33 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=91118 bytes.received=460094531 bytes.decompressed=0
Tue Oct 27 21:20:38 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=102145 bytes.received=516585629 bytes.decompressed=0
Tue Oct 27 21:20:43 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=105727 bytes.received=534320643 bytes.decompressed=0
Tue Oct 27 21:20:48 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=103850 bytes.received=526447291 bytes.decompressed=0
Tue Oct 27 21:20:53 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=102585 bytes.received=519250454 bytes.decompressed=0


**** latency for pssh -h /tmp/ips -t 3600 -p 200 -P "time /tmp/tcpflood -c20 -m100000 -r -d10000 -P129 -D -t 10.34.117.153 -R 1 -S 0"
38.204
43.918
45.657
47.136
47.169
47.256
47.306
47.552
47.972
49.297


*** stats with octate counting

Tue Oct 27 21:26:59 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=0 bytes.received=0 bytes.decompressed=0
Tue Oct 27 21:27:04 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=0 bytes.received=0 bytes.decompressed=0
Tue Oct 27 21:27:10 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=3550 bytes.received=17187920 bytes.decompressed=0
Tue Oct 27 21:27:15 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=107507 bytes.received=540373491 bytes.decompressed=0
Tue Oct 27 21:27:20 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=111172 bytes.received=561962081 bytes.decompressed=0
Tue Oct 27 21:27:25 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=107006 bytes.received=543170829 bytes.decompressed=0
Tue Oct 27 21:27:30 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=114151 bytes.received=575287884 bytes.decompressed=0
Tue Oct 27 21:27:35 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=112556 bytes.received=566777748 bytes.decompressed=0
Tue Oct 27 21:27:40 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=112117 bytes.received=565782445 bytes.decompressed=0
Tue Oct 27 21:27:45 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=108962 bytes.received=551616650 bytes.decompressed=0
Tue Oct 27 21:27:50 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=110280 bytes.received=557155126 bytes.decompressed=0
Tue Oct 27 21:27:55 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=107973 bytes.received=545592782 bytes.decompressed=0
Tue Oct 27 21:28:00 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=94548 bytes.received=476255888 bytes.decompressed=0
Tue Oct 27 21:28:05 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=95275 bytes.received=481822825 bytes.decompressed=0
Tue Oct 27 21:28:10 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=96591 bytes.received=490263548 bytes.decompressed=0
Tue Oct 27 21:28:15 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=96105 bytes.received=487132474 bytes.decompressed=0
Tue Oct 27 21:28:20 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=94608 bytes.received=478366724 bytes.decompressed=0
Tue Oct 27 21:28:25 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=96010 bytes.received=485047986 bytes.decompressed=0
Tue Oct 27 21:28:30 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=95268 bytes.received=479963258 bytes.decompressed=0
Tue Oct 27 21:28:35 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=96290 bytes.received=486733010 bytes.decompressed=0
Tue Oct 27 21:28:40 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=93781 bytes.received=472543150 bytes.decompressed=0
Tue Oct 27 21:28:45 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=93950 bytes.received=473332175 bytes.decompressed=0
Tue Oct 27 21:28:50 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=90335 bytes.received=456605246 bytes.decompressed=0
Tue Oct 27 21:28:55 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=97829 bytes.received=496778238 bytes.decompressed=0
Tue Oct 27 21:29:00 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=107820 bytes.received=547579829 bytes.decompressed=0
Tue Oct 27 21:29:05 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=112003 bytes.received=567745673 bytes.decompressed=0
Tue Oct 27 21:29:10 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=109921 bytes.received=555909390 bytes.decompressed=0
Tue Oct 27 21:29:15 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=110233 bytes.received=560283251 bytes.decompressed=0
Tue Oct 27 21:29:20 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=110779 bytes.received=560270018 bytes.decompressed=0
Tue Oct 27 21:29:25 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=113179 bytes.received=573136065 bytes.decompressed=0
Tue Oct 27 21:29:30 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=93731 bytes.received=474099443 bytes.decompressed=0
Tue Oct 27 21:29:35 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=93108 bytes.received=470591425 bytes.decompressed=0
Tue Oct 27 21:29:40 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=93094 bytes.received=470888388 bytes.decompressed=0
Tue Oct 27 21:29:45 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=91553 bytes.received=461952656 bytes.decompressed=0
Tue Oct 27 21:29:50 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=94905 bytes.received=480134980 bytes.decompressed=0
Tue Oct 27 21:29:55 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=93694 bytes.received=472786576 bytes.decompressed=0
Tue Oct 27 21:30:00 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=96117 bytes.received=482963179 bytes.decompressed=0
Tue Oct 27 21:30:05 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=93289 bytes.received=470395587 bytes.decompressed=0
Tue Oct 27 21:30:10 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=92771 bytes.received=467496032 bytes.decompressed=0
Tue Oct 27 21:30:15 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=97002 bytes.received=488305458 bytes.decompressed=0
Tue Oct 27 21:30:20 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=92759 bytes.received=467824763 bytes.decompressed=0
Tue Oct 27 21:30:25 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=95756 bytes.received=481868803 bytes.decompressed=0
Tue Oct 27 21:30:30 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=110138 bytes.received=554647467 bytes.decompressed=0
Tue Oct 27 21:30:35 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=110396 bytes.received=557068625 bytes.decompressed=0
Tue Oct 27 21:30:40 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=110791 bytes.received=561153954 bytes.decompressed=0
Tue Oct 27 21:30:45 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=110565 bytes.received=559514727 bytes.decompressed=0
Tue Oct 27 21:30:50 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=112974 bytes.received=571181305 bytes.decompressed=0
Tue Oct 27 21:30:56 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=113964 bytes.received=577987743 bytes.decompressed=0
Tue Oct 27 21:31:01 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=113372 bytes.received=574744470 bytes.decompressed=0
Tue Oct 27 21:31:06 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=101530 bytes.received=514706943 bytes.decompressed=0
Tue Oct 27 21:31:11 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=95877 bytes.received=485702521 bytes.decompressed=0
Tue Oct 27 21:31:16 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=94236 bytes.received=478504456 bytes.decompressed=0
Tue Oct 27 21:31:21 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=97391 bytes.received=493846190 bytes.decompressed=0
Tue Oct 27 21:31:26 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=104129 bytes.received=528120704 bytes.decompressed=0
Tue Oct 27 21:31:31 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=127247 bytes.received=644711274 bytes.decompressed=0
Tue Oct 27 21:31:36 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=126909 bytes.received=642799651 bytes.decompressed=0
Tue Oct 27 21:31:41 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=125391 bytes.received=636229212 bytes.decompressed=0
Tue Oct 27 21:31:46 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=124620 bytes.received=632967356 bytes.decompressed=0
Tue Oct 27 21:31:51 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=124010 bytes.received=628428619 bytes.decompressed=0
Tue Oct 27 21:31:56 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=125074 bytes.received=633789427 bytes.decompressed=0
Tue Oct 27 21:32:01 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=129476 bytes.received=656329787 bytes.decompressed=0
Tue Oct 27 21:32:06 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=114276 bytes.received=578089571 bytes.decompressed=0
Tue Oct 27 21:32:11 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=116490 bytes.received=592428500 bytes.decompressed=0
Tue Oct 27 21:32:16 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=112663 bytes.received=570278034 bytes.decompressed=0
Tue Oct 27 21:32:21 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=114867 bytes.received=580913160 bytes.decompressed=0
Tue Oct 27 21:32:26 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=118939 bytes.received=602288703 bytes.decompressed=0
Tue Oct 27 21:32:31 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=118436 bytes.received=599864821 bytes.decompressed=0
Tue Oct 27 21:32:36 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=114969 bytes.received=581557937 bytes.decompressed=0
Tue Oct 27 21:32:41 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=113271 bytes.received=571628815 bytes.decompressed=0
Tue Oct 27 21:32:46 2015: imptcp(*/13514/IPv4): origin=imptcp submitted=115195 bytes.received=581631930 bytes.decompressed=0

**** latency for pssh -h /tmp/ips -t 3600 -p 200 -P "time /tmp/tcpflood -c20 -m100000 -r -d10000 -P129 -D -t 10.34.117.153 -R 1 -S 0 -O"
41.365
41.443
41.761
40.771
42.675
41.879
44.601
43.787
44.081
45.223

janmejay commented:
Successfully posted the detail, copy it to a non-wraping buffer for readability :-p

rgerhards commented:
I'll work on this on an interim branch, as I first need to fix the conflicts that happend due code changes since this PR was filed.

rgerhards commented on tests/chkseq.c:
```diff
@@ -172,9 +187,18 @@ int main(int argc, char *argv[])
+						if (bAnticipateTruncation == 1) {
+							if (edLen < strlen(edBuf)) {
+								printf("extra data length specified %d, but actually is %ld in record %d"
+									   " (truncation was anticipated, but payload should have been smaller than data-length, not smaller)\n",
```
Is the error message correct? It sounds like small/larger was mixed.

rgerhards commented on tests/chkseq.c:
```diff
@@ -118,9 +124,18 @@ int main(int argc, char *argv[])
+				if (bAnticipateTruncation == 1) {
+					if (edLen < strlen(edBuf)) {
+						printf("extra data length specified %d, but actually is %ld in record %d"
+							   " (truncation was anticipated, but payload should have been smaller than data-length, not smaller)\n",
```
Is the error message correct? It sounds like small/larger was mixed.

rgerhards commented:
@janmejay The imptcp_nonProcessingPoller.sh is very verbose and spams the testbench log. Can we do something against this?

janmejay commented:
I am not sure how verbosity manifests. Can you please point me to a link
where I can see it?

Fixed the incorrect error message and added commit to PR.

On Thu, Jan 28, 2016 at 4:17 PM, Rainer Gerhards <notifications@github.com>
wrote:

> @janmejay <https://github.com/janmejay> The imptcp_nonProcessingPoller.sh
> is very verbose and spams the testbench log. Can we do something against
> this?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rsyslog/rsyslog/pull/567#issuecomment-176119852>.
>



-- 
Regards,
Janmejay
http://codehunk.wordpress.com

rgerhards commented:
On the verbosity: have a look here https://travis-ci.org/rgerhards/rsyslog/jobs/105380327

basically, there are many lines of

```rsyslogd: received oversize message: size is 54692 bytes, max msg size is 10240, truncating... [v8.17.0.master] ```

janmejay commented:
Will fix this by tomorrow.

--
Regards,
Janmejay

PS: Please blame the typos in this mail on my phone's uncivilized soft
keyboard sporting it's not-so-smart-assist technology.

On Jan 28, 2016 9:01 PM, "Rainer Gerhards" <notifications@github.com> wrote:

> On the verbosity: have a look here
> https://travis-ci.org/rgerhards/rsyslog/jobs/105380327
>
> basically, there are many lines of
>
> rsyslogd: received oversize message: size is 54692 bytes, max msg size is
> 10240, truncating... [v8.17.0.master]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rsyslog/rsyslog/pull/567#issuecomment-176237372>.
>

janmejay commented:
Im clueless why it does that. I can silence it by simply taking stderr to
/dev/null, but that error log is actually not supposed to be generated.

It seems tcpflood is doing something its not supposed to. The message sizes
it posts are bounded by MAX_EXTRADATA_LEN (which is 100k).

Also, it seems one of gcc and clang builds produced 18004 such warnings and
another produced 17990, which shows statistically even distribution. Max
size ~100k, acceptable size ~10k, 10k is 10% of 100k, we sent 20k messages,
10% of that is 2k, 20k - 2k = 18k, which is the number of messages that
should statistically violate max-size limit.

All evidence seems to point to tcpflood messing up extra-data limit
prescribed in command line and going with the buffer-size (and then
randomization doing its thing very well), but I can't reproduce it, neither
on my env, nor on Ubuntu 12.04 LTS.

Tried running it with valgrind hoping for luck, but didn't happen.

I think I'll add some instrumentation to confirm it is tcpflood.

On Thu, Jan 28, 2016 at 9:47 PM, singh.janmejay <singh.janmejay@gmail.com>
wrote:

> Will fix this by tomorrow.
>
> --
> Regards,
> Janmejay
>
> PS: Please blame the typos in this mail on my phone's uncivilized soft
> keyboard sporting it's not-so-smart-assist technology.
>
> On Jan 28, 2016 9:01 PM, "Rainer Gerhards" <notifications@github.com>
> wrote:
>
>> On the verbosity: have a look here
>> https://travis-ci.org/rgerhards/rsyslog/jobs/105380327
>>
>> basically, there are many lines of
>>
>> rsyslogd: received oversize message: size is 54692 bytes, max msg size is
>> 10240, truncating... [v8.17.0.master]
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/rsyslog/rsyslog/pull/567#issuecomment-176237372>.
>>
>


-- 
Regards,
Janmejay
http://codehunk.wordpress.com

janmejay commented:
@rgerhards can you please push this patch on top of that branch? it'll probably give us some clue of what is going on.

rgerhards commented:
I can, but unfortunately only tomorrow. Alternatively, you can just mine
back into yours, result would be the same from Travis point of view.
Rainer

Sent from phone, thus brief.
Am 29.01.2016 23:00 schrieb "Janmejay Singh" <notifications@github.com>:

> @rgerhards <https://github.com/rgerhards> can you please push this patch
> on top of that branch? it'll probably give us some clue of what is going on.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rsyslog/rsyslog/pull/567#issuecomment-176988804>.
>

janmejay commented:
Of course, don't know why I didn't think of it.

When are you back?

--
Regards,
Janmejay

PS: Please blame the typos in this mail on my phone's uncivilized soft
keyboard sporting it's not-so-smart-assist technology.

On Jan 30, 2016 3:32 AM, "Rainer Gerhards" <notifications@github.com> wrote:

> I can, but unfortunately only tomorrow. Alternatively, you can just mine
> back into yours, result would be the same from Travis point of view.
> Rainer
>
> Sent from phone, thus brief.
> Am 29.01.2016 23:00 schrieb "Janmejay Singh" <notifications@github.com>:
>
> > @rgerhards <https://github.com/rgerhards> can you please push this patch
> > on top of that branch? it'll probably give us some clue of what is going
> on.
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <https://github.com/rsyslog/rsyslog/pull/567#issuecomment-176988804>.
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rsyslog/rsyslog/pull/567#issuecomment-176989202>.
>

janmejay commented:
Ok, this fixes the verbosity problem.

There was no bug in tcpflood, the
test imptcp_veryLargeOctateCountedMessages.sh was
titled imptcp_nonProcessingPoller.sh (in that first echo we do of test name
and description). We never noticed that nonProcessingPoller tests are
running twice (the other one was veryLargeOctateCountedMessages test, which
is supposed to truncate messages as we send upto 100k length extra-data).

So we now have a start-slient, which /dev/nulls stderr.

Also made a minor change in tcpflood, which allows it to adjust
sent-message-count reporting rate (keeping overall number of reported
data-points the same).

This is now ready for merge.

On Sat, Jan 30, 2016 at 9:48 AM, singh.janmejay <singh.janmejay@gmail.com>
wrote:

> Of course, don't know why I didn't think of it.
>
> When are you back?
>
> --
> Regards,
> Janmejay
>
> PS: Please blame the typos in this mail on my phone's uncivilized soft
> keyboard sporting it's not-so-smart-assist technology.
>
> On Jan 30, 2016 3:32 AM, "Rainer Gerhards" <notifications@github.com>
> wrote:
>
>> I can, but unfortunately only tomorrow. Alternatively, you can just mine
>> back into yours, result would be the same from Travis point of view.
>> Rainer
>>
>> Sent from phone, thus brief.
>> Am 29.01.2016 23:00 schrieb "Janmejay Singh" <notifications@github.com>:
>>
>> > @rgerhards <https://github.com/rgerhards> can you please push this
>> patch
>> > on top of that branch? it'll probably give us some clue of what is
>> going on.
>> >
>> > —
>> > Reply to this email directly or view it on GitHub
>> > <https://github.com/rsyslog/rsyslog/pull/567#issuecomment-176988804>.
>> >
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/rsyslog/rsyslog/pull/567#issuecomment-176989202>.
>>
>


-- 
Regards,
Janmejay
http://codehunk.wordpress.com

rgerhards commented:
Args, I am back, but I forgot today. Shame on me...

Sent from phone, thus brief.
Am 30.01.2016 05:18 schrieb "Janmejay Singh" <notifications@github.com>:

> Of course, don't know why I didn't think of it.
>
> When are you back?
>
> --
> Regards,
> Janmejay
>
> PS: Please blame the typos in this mail on my phone's uncivilized soft
> keyboard sporting it's not-so-smart-assist technology.
>
> On Jan 30, 2016 3:32 AM, "Rainer Gerhards" <notifications@github.com>
> wrote:
>
> > I can, but unfortunately only tomorrow. Alternatively, you can just mine
> > back into yours, result would be the same from Travis point of view.
> > Rainer
> >
> > Sent from phone, thus brief.
> > Am 29.01.2016 23:00 schrieb "Janmejay Singh" <notifications@github.com>:
> >
> > > @rgerhards <https://github.com/rgerhards> can you please push this
> patch
> > > on top of that branch? it'll probably give us some clue of what is
> going
> > on.
> > >
> > > —
> > > Reply to this email directly or view it on GitHub
> > > <https://github.com/rsyslog/rsyslog/pull/567#issuecomment-176988804>.
> > >
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <https://github.com/rsyslog/rsyslog/pull/567#issuecomment-176989202>.
> >
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rsyslog/rsyslog/pull/567#issuecomment-177062484>.
>
